### ISSUE TITLE
[Question] Routing from two lane to one lane

### ISSUE THREAD

**Issue by dymons** [fzi-forschungszentrum-informatik/Lanelet2#64]
When laying a path (using function getRoute(...)) an error occurs when another lanelet is created from a two-lane lane to rebuild.
In the first picture, from a two-lane lane to a single-lane lane, there is only one lanelet lane change, and the path is paved.
![1](https://user-images.githubusercontent.com/20445903/70889279-4c8c5280-1ff3-11ea-93da-2f685d8a75ef.png)
When you add a second, neighboring lanelet'a, the path ceases to be laid and an error occurs.
![2](https://user-images.githubusercontent.com/20445903/70889296-56ae5100-1ff3-11ea-9599-84ecf5a01611.png)
![Screenshot from 2019-12-16 10-48-58](https://user-images.githubusercontent.com/20445903/70889310-5e6df580-1ff3-11ea-9798-9f41087e4bff.png)
What can be the reason and how to solve it? There are similar situations in the map and there are no such problems.

**Comment by poggenhans** [comment]
Hm, strange. Thanks for reporting. I cannot see why your map should trigger this assertion.

We are currently about to refactor the internal representation of the `Route` object and how it is created. I pushed the current state to the branch `routing_update`. Can you test if that state fixes your problem?

**Comment by dymons** [comment]
Yes, it helped!
![rviz_screenshot_2019_12_16-13_01_45(1)](https://user-images.githubusercontent.com/20445903/70898037-ed373e00-2004-11ea-87e3-d3be4811c28d.png)

**Comment by dymons** [comment]
I decided to test your `routing_update` branch, and found a strange error when calling getRoute().
On the master branch
![rviz_screenshot_2019_12_16-14_48_51](https://user-images.githubusercontent.com/20445903/70904803-6178de00-2013-11ea-85f9-ca3cd7a62ff6.png)
On the `routing_update` branch
![rviz_screenshot_2019_12_16-14_40_20](https://user-images.githubusercontent.com/20445903/70904796-5b82fd00-2013-11ea-8469-49b0d0ab5408.png)
Can I help you debug the new functionality or will it be easier to throw you our map for debugging?

**Comment by poggenhans** [comment]
It seems as if the shortestPath that forms the basis of the Route is no longer correct. I added our latest changes to the shortestPath calculation. Can you pull and try again?

If it still doesn't work, the problem must be somewhere deeper. Having your map for debugging this would be very helpful.

**Comment by dymons** [comment]
I deleted the last comment because I thought I was wrong. But I checked this moment, and there really is a strange behavior.
This problem occurred when calling the getRouteVia(4242, 4211, 13079).
On a branch `master` correctly.
![rviz_screenshot_2019_12_17-11_20_06(1)](https://user-images.githubusercontent.com/20445903/70983148-4668a580-20c9-11ea-9918-957eeb62937c.png)
On a branch `routing_update` not correctly.
![rviz_screenshot_2019_12_17-11_23_04(1)](https://user-images.githubusercontent.com/20445903/70983180-554f5800-20c9-11ea-9abe-4b215f12aff8.png)
I do not want to spread the map among a large number of people. Tell me your email, the map will be sent by mail.
